app/src/main/java/com/nervousfish/nervousfish/data_objects/RSAKey.java

```diff
@@ -53,9 +53,9 @@ public RSAKey(final Map<String, String> map) throws IllegalArgumentException {
      *
      * @param key An {@link RSAKey}.
      * @return The value for the exponent attribute of the {@code key}.
-     * @throws Exception If the provided {@link IKey} is not a {@link RSAKey}.
+     * @throws IllegalArgumentException If the provided {@link IKey} is not a {@link RSAKey}.
```

Comment:
You don't have the document exceptions, because it should be clear from the rest of the documentation that a certain exception will be thrown if they use the method incorrectly. Nevertheless, now it's there it doesn't make sense to remove it